docs/content/Overlay.mdx

```diff
@@ -0,0 +1,80 @@
+- If the `Overlay` should also have a longer description, use `aria-describedby`
+
+## Default example
+
+```javascript live noinline
+const Demo = () => {
+  // you must manage your own open state
+  const [isOpen, setIsOpen] = React.useState(false)
+  const noButtonRef = React.useRef(null)
+  const anchorRef = React.useRef(null)
+  return (
+    <>
+      <Button ref={anchorRef} onClick={() => setIsOpen(!isOpen)}>
+        open overlay
+      </Button>
+      {/* be sure to conditionally render the Overlay. This helps with performance and is required. */}
+      {isOpen &&
+        <Overlay
+          anchorRef={anchorRef}
+          initialFocusRef={noButtonRef}
+          returnFocusRef={anchorRef}
+          ignoreClickRefs={[anchorRef]}
+          onEscape={() => setIsOpen(!isOpen)}
+          onClickOutside={() => setIsOpen(false)}
+          aria-labelledby="title"
+        >
+          <Flex flexDirection="column" p={2}>
+            <Text id="title">Are you sure you would like to delete this item?</Text>
+            <Button >yes</Button>
+            <Button ref={noButtonRef}>no</Button>
+          </Flex>
+        </Overlay>
+      }
+
+    </>
+  )
+}
+
+render(<Demo/>)
+```
+
+## System props
+
+`Overlay` gets `COMMON` system props. Read our [System Props](/system-props) doc page for a full list of available props.
+
+## Component props
+
+| Name | Type   | Default | Description                         |
+| :--- | :----- | :-----: | :---------------------------------- |
+| positionSettings | See the [`PositionSettings interface`]() section of the `anchoredPosition` docs | | Optional. Settings used to position the `Overlay`. If none are provided, `Overlay` is positioned on the bottom right of the `anchorRef`. |
+| positionDeps | `React.DependencyList` | | Optional. If defined, the position of the `Overlay` will only be recalulated when one of the dependencies in this array changes. |
+| ignoreClickRefs   | `React.RefObject []` |  | Optional. An array of ref objects to ignore clicks on in the `onOutsideClick` behavior. This is often used to ignore clicking on the element that toggles the open/closed state for the `Overlay` to prevent the `Overlay` from being toggled twice. |
+| inititalFocusRef  | `React.RefObject` |  | Optional. Ref for the element to focus when the `Overlay` is opened. If nothing is provided, the first focusable element in the `Overlay` body is focused.  |
+| anchorRef | `React.RefObject` |  | Required. Element the `Overlay` should be anchored to. |
+| returnFocusRef  | `React.RefObject` |  | Required. Ref for the element to focus when the `Overlay` is closed.  |
+| onClickOutside  | `function` |  | Required. Function to call when clicking outside of the `Overlay`. Typically this function sets the `Overlay` visibility state to `false`.  |
+| onEscape  | `function` |  | Required. Function to call when user presses `Escape`. Typically this function sets the `Overlay` visibility state to `false`.  |
```

Comment:
Can these be consolidated to one prop? I'm thinking `onDismiss: (method: "click-away" | "escape") => void`.

Reply:
I was trying to separate these callbacks from assuming that the user would want to dismiss when called + I think keeping them as separate props feels a bit more declarative/cleaner